tests/test_utils.py

```diff
@@ -239,3 +239,53 @@ def test_get_instance_token_file_name():
     expected = os.path.join(os.path.expanduser('~'), '.osbs', 'spam.token')
 
     assert get_instance_token_file_name('spam') == expected
+
+@pytest.mark.parametrize(('labels', 'fnc', 'expect'), [
+        ({},
+         ("get_label_only", Labels.LABEL_TYPE_COMPONENT),
+         "com.redhat.component"),
+        ({},
+         ("get_label_only", "new_label"),
+         "new_label"),
+        ({},
+         ("get_label_only", ""),
+         ""),
+        ({"Name" : "old",
+          "name" : "new"},
+         ("get_label_only", Labels.LABEL_TYPE_NAME),
+         "name"),
+        ({"Name" : "old"},
+         ("get_label_only", Labels.LABEL_TYPE_NAME),
+         "Name"),
+        ({},
+         ("get_new_names_by_old", None),
+         {"Vendor": "vendor", "Name": "name", "Build_Host": "com.redhat.build-host",
+          "Version": "version", "Architecture": "architecture",
+          "Release": "release", "BZComponent": "com.redhat.component",
+          "Authoritative_Registry": "authoritative-source-url"}),
+        ({"Name" : "old",
+          "name" : "new"},
+          ("get_label_and_value", Labels.LABEL_TYPE_NAME),
+          ("name", "new")),
+        ({},
+          ("get_label_and_value", "doest_exist"),
+          KeyError),
```

Comment:
"doest_exist" is not a valid key -- calling this method with some invalid key is not defined behaviour and we should not test it.